File: frontend/src/types/subreddit.ts
export interface Subreddit {
  name: string;
  display_name: string;
  description: string | null;
  subscribers: number | null;
  active_users: number | null;
  posts_per_day: number | null;
  comments_per_day: number | null;
  growth_rate: number | null;
  relevance_score: number | null;
  created_at: string | null;
  updated_at: string | null;
  last_updated: string | null;
}

export interface SubredditResponse {
  data: Subreddit[];
  status: number;
  message?: string;
} 

File: backend/app/schemas/subreddit.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict


class SubredditResponse(BaseModel):
    name: str
    display_name: str
    description: Optional[str] = None
    subscribers: Optional[int] = None
    active_users: Optional[int] = None
    posts_per_day: Optional[float] = None
    comments_per_day: Optional[float] = None
    growth_rate: Optional[float] = None
    relevance_score: Optional[float] = None
    created_at: Optional[datetime] = None
    updated_at: Optional[datetime] = None

    model_config = ConfigDict(
        from_attributes=True,
        json_encoders={
            datetime: lambda dt: dt.isoformat() if dt else None
        }
    )


class KeywordSuggestionResponse(BaseModel):
    keyword: str
    score: float  # Relevance score
    subreddit_count: Optional[int] = None  # Number of subreddits matching this keyword

    class Config:
        from_attributes = True 